### PR TITLE
feat(perf-issues): Account for multiple file-io spans

### LINKED ISSUE #42074: Update the performance detector to take the total blocking thread time rather than an individual spans blocking time
Example event with multiple: https://gist.github.com/wmak/540ee2305a9305abe7d28fa8db620dbd

### PR DESCRIPTION
**Pull request by wmak**
- Updates the FileIOMainThreadDetector to account for multiple spans with the same parent. If the total span time exceeds the threshold then make all of them part of the performance issue
- For now just joining all of the call stacks together as the fingerprint. This has the benefit of the original fingerprints being unchanged, but may not be sufficient for the multiple span case
- Closes #42074

### CODE REVIEW

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -496,6 +497,29 @@ def contains_complete_query(span: Span, is_source: Optional[bool] = False) -> bo
         return query and not query.endswith("...")
 
 
+def total_span_time(span_list: List[Dict[str, Any]]) -> float:
```
Comment (gggritso) [inline src/sentry/utils/performance_issues/performance_detection.py:500]: I just burned myself pretty good with assuming that a `float` that a function was returning was in milliseconds, but it was seconds, so I'm scared of this `-> float`. I think a `timedelta` is better, but that's not a strong opinion!

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 2]
```diff
@@ -1421,30 +1446,42 @@ def _deobfuscate_function(self, frame):
...
+            settings_for_span = self.settings_for_span(span_list[0])
             if not settings_for_span:
                 return
 
-            op, span_id, op_prefix, span_duration, settings = settings_for_span
-            if span_duration.total_seconds() * 1000 > settings["duration_threshold"]:
-                fingerprint = self._fingerprint(span)
+            _, _, _, _, settings = settings_for_span
+            if total_duration >= settings["duration_threshold"]:
+                fingerprint = self._fingerprint(span_list)
```
Comment (k-fish) [inline src/sentry/utils/performance_issues/performance_detection.py:1464]: 👍

**Review by gggritso — APPROVED** [review]
Makes sense! The fixtures have a lot of unnecessary data in them, but is that hurting anything, really? Probably not! ✨ LGTM

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 3]
```diff
@@ -1421,30 +1446,42 @@ def _deobfuscate_function(self, frame):
...
-            call_stack_strings.append(f"{module}.{function}")
-        call_stack = ".".join(call_stack_strings).encode("utf8")
+        overall_stack = []
+        for span in span_list:
+            for item in span.get("data", {}).get("call_stack", []):
+                module = self._deobfuscate_module(item.get("module", ""))
+                function = self._deobfuscate_function(item)
+                call_stack_strings.append(f"{module}.{function}")
+            overall_stack.append(".".join(call_stack_strings))
+        call_stack = "-".join(overall_stack).encode("utf8")
```
Comment (k-fish) [inline src/sentry/utils/performance_issues/performance_detection.py:1484]: Just to be clear, have you been checking the performance of this new detector in our metrics? stacktraces are pretty expensive to muck with, you might want to wrap this in a `timer` if you aren't tracking this elsewhere.

**Review by k-fish — APPROVED** [review]
Nice!

**Review by shruthilayaj — APPROVED** [review]

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 3 above)
Comment (wmak) [reply src/sentry/utils/performance_issues/performance_detection.py:1484]: not yet, we're still waiting on the SDKs to release with the new blocked_main_thread attribute but i'll keep an eye out on perf once that's ready

**Pull request merged**